scvi/module/_mrdeconv.py

```diff
@@ -329,7 +349,7 @@ def get_ct_specific_expression(
             integer for cell types
         """
         # cell-type specific gene expression, shape (minibatch, celltype, gene).
-        beta = torch.nn.functional.softplus(self.beta)  # n_genes
+        beta = torch.exp(self.beta)  # n_genes
```

Comment:
same question

Reply:
Beta is regularized by var and the use of softplus leads to values in the range of 1e-3 (smaller change of beta lead to bigger change in softplus(beta), whereas exp leads to values close to 1. I prefer this but am open to keep it softplus.

Reply:
Oh, we are speaking about different beta. I'm speaking about the output of softplus being 1e-3.As the rate of softplus is not constant, our variance regularization leads to shrinkage of the betas towards -4, which is the region where softplus is an exponential function. It's not an issue performance-wise and I can switch back to softplus. However, I don't really like this property (beta close to 0 is more reasonable to me for interpretation).

Reply:
I don't expect users to look at the betas so unclear if it all matters really. I just wanted to avoid numerical stability. Did you lower the learning rate to 0.003 because otherwise you have numerical instabilities? If yes, I'd lean towards returning to softplus. If you have no numerical instabilities and similar results we can keep exp.

Reply:
No it's not unstable also with lr=0.01. Lower leaning rate led to slightly lower losses. I am pretty convinced the rather large weight on torch.var(beta) ensures there is no instability.